Documentation/swagger-extensions.md

```diff
@@ -0,0 +1,180 @@
+* [x-ms-enum](#x-ms-enum) - additional metadata for enums
+* [x-ms-parameter-grouping](#x-ms-parameter-grouping) - groups method parameters in generated clients
+* [x-ms-paths](#x-ms-paths) - alternative to [Paths Object](https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#pathsObject) that allows [Path Item Object](https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#pathItemObject) to have query parameters for non pure REST APIs
+* x-ms-client-name - *not currently implemented*
+* [x-ms-external](#x-ms-external) - allows specific [Definition Objects](https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#definitionsObject) to not be excluded from code generation
+
+## Microsoft Azure Extensions
+* x-ms-odata
+* x-ms-pageable
+* x-ms-long-running-operation
+* x-ms-azure-resource
+* x-ms-discriminator-value 
+* x-ms-request-id
+* x-ms-client-request-id
+
+## x-ms-skip-url-encoding
+By default, `path` parameters will be URL-encoded automatically. This is a good default choice for user-provided values. This is not a good choice when the parameter is provided from a source where the value is known to be URL-encoded. The URL encoding is NOT an idempotent operation. For example, the percent character "%" is URL-encoded as "%25". If the parameter is URL-encoded again, "%25" becomes "%2525". Mark parameters where the source is KNOWN to be URL-encoded to prevent the automatic encoding behavior.
+
+**Parent element**: [Parameter Object](https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#parameterObject)
+
+**Schema**: 
+`true|false`
+
+**Example**:
+```js
+"parameters": [
+  {
+    "name": "databaseName",
+    "in": "path",
+    "type": "string",
+    "required": true,
+    "x-ms-skip-url-encoding": true
+  }
+]
+```
+
+## x-ms-enum
+Enum definitions in Swagger indicate that only a particular set of values may be used for a property or parameter. When the property is represented on the wire as a string, it would be a natural choice to represent the property type in C# as an enum. However, not all enumeration values should necessarily be represented as C# enums - there are additional considerations, such as how often expected values might change, since adding a new value to a C# enum is a breaking change requiring an updated API version. Additionally, there is some metadata that is required to create a useful C# enum, such as a descriptive name, which is not represented in swagger. For this reason, enums are not automatically turned into enum types in C# - instead they are rendered in the documentation comments for the property or parameter to indcate allowed values. To indicate that an enum will rarely change and that C# enum semantics are desired, use the `x-ms-enum` exension.
+
+In C#, an enum type is generated and is declared as the type of the related request/response object. The enum is serialized as the string expected by the REST API.
```

Comment:
Do generators for other languages currently ignore x-ms-enum? Either way, we should call it out here.